### DISCUSSION TITLE
nx.all_simple_paths with data restraint request

### DISCUSSION THREAD

**Discussion by pastastical** [networkx/networkx#4442] [Q&A] [answered]
Hi!

Hopefully this is a good place to ask a question.

I'm trying to find simple paths between all nodes while restricting the possible next paths to a range of data values in the current path. I'm using datetime, but a simplified example would be:

Edges in G:
( A, B ,val=1 )
( B, C, val = 5 )
( B, D, val = 8 )
( B, E, val = 2 )
( B, F, val = 12 )
( E, G, val = 3 )
(E, H, val = 6)
I'd like to say nx.all_simple_paths(G,iterative_edge_data_restraint=2) and have the results [ (A,B), (B,E), (E,G) ] because The edge data for edge(A, B) is 1 and data for edge(B, E) is the only edge from B which also has a data value within 2 units of edge(A, B)'s data value.
 
Also edge(E,G) is included because it has a data value within two units of edge(B, E)'s data value

I don't know how to do that from the documentation i can find.

**Comment by dschult** [comment, marked as answer]
You aren't finding paths, you are finding edges... Perhaps use something like `bfs_edges` and only accepting the ones within multiple 2 of the previous edge value.  You'll have to keep track of the edge value and the multiple in your own code. There isn't a built-in way to do what you are looking for. 

Warning: not tested...
```python
CUTOFF = 2
val = 0
for u, v in nx.bfs_edges(G, source='A'):
    newval = G[u][v]['val']
    if abs(newval-val)<CUTOFF:
        continue
    yield (u, v)
    val = G[u][v]['val']
```
